### ISSUE TITLE
Datatrack send errors are not being logged

### ISSUE THREAD

**Issue by EdGallimore** [twilio/twilio-video.js#1509]
- [X] I have verified that the issue occurs with the latest twilio-video.js release and is not marked as a known issue in the [CHANGELOG.md](https://github.com/twilio/twilio-video.js/blob/master/CHANGELOG.md).
 - [X] I reviewed the [Common Issues](https://github.com/twilio/twilio-video.js/blob/master/COMMON_ISSUES.md) and open GitHub issues and verified that this report represents a potentially new issue.
 - [X] I verified that the [Quickstart](https://github.com/twilio/video-quickstart-js) application works in my environment.
 - [X] I am not sharing any [Personally Identifiable Information (PII)](https://www.twilio.com/docs/glossary/what-is-personally-identifiable-information-pii)
       or sensitive account information (API keys, credentials, etc.) when reporting this issue.

(Closed and consolidated with issue: https://github.com/twilio/twilio-video.js/issues/1510)

**Behaviour**
Our application connects to a room, publishes a dataTrack and immediately attempts to send a message to the other participants in the room.  This message is not received by the other participants. No error is thrown when calling the send method is called

**Observation**
On investigation it seems possible the send method might be throwing an error (given the state of the connection). But it appears that the DataTrackSender twillio-video implementation swallows errors (rather than handling or even logging them to the logger)
Code snippet from twilio-video below

```
 send(data) {
    this._dataChannels.forEach(dataChannel => {
      try {
        dataChannel.send(data);
      } catch (error) {
        // Do nothing.
      }
    });
    this._clones.forEach(clone => {
      try {
        clone.send(data);
      } catch (error) {
        // Do nothing.
      }
    });
    return this;
  }
```

**Software versions:**
Chrome v89 on MacOs
twilio-video.js 2.15.1
react 17.0.2

**Comment by charliesantos** [comment]
@EdGallimore the repro steps seems very similar to this other issue you filed #1510 . Can you combine both if possible?

**Comment by EdGallimore** [comment]
(Closed and consolidated with issue: #1510)
